pkg/products/cloudresources/reconciler_test.go

```diff
@@ -415,3 +415,215 @@ func TestReconciler_setPlatformStrategyName(t *testing.T) {
+				Reconciler: resources.NewReconciler(&marketplace.MarketplaceInterfaceMock{}).
+					WithProductDeclaration(marketplace.ProductDeclaration{}),
+				recorder: nil,
+			},
+			args: args{
+				client: moqclient.NewSigsClientMoqWithScheme(scheme, &configv1.Infrastructure{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cluster",
+					},
+					Status: configv1.InfrastructureStatus{
+						PlatformStatus: &configv1.PlatformStatus{
+							Type: configv1.GCPPlatformType,
+						},
+					},
+				}),
+				maintenanceDay:  "Monday",
+				maintenanceHour: "2",
+				ctx:             context.TODO(),
+			},
+			want:    integreatlyv1alpha1.PhaseFailed,
+			wantErr: true,
+		},
+		{
+			name: "error when incorrect values entered for maintenanceHour",
+			fields: fields{
+				Config: nil,
+				ConfigManager: &config.ConfigReadWriterMock{
+					GetOperatorNamespaceFunc: func() string {
+						return "redhat-rhoam-operator"
+					},
+				},
+				installation: nil,
+				mpm:          nil,
+				log:          getLogger(),
+				Reconciler: resources.NewReconciler(&marketplace.MarketplaceInterfaceMock{}).
+					WithProductDeclaration(marketplace.ProductDeclaration{}),
+				recorder: nil,
+			},
+			args: args{
+				client: moqclient.NewSigsClientMoqWithScheme(scheme, &configv1.Infrastructure{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cluster",
+					},
+					Status: configv1.InfrastructureStatus{
+						PlatformStatus: &configv1.PlatformStatus{
+							Type: configv1.GCPPlatformType,
+						},
+					},
+				}),
+				maintenanceDay:  "4",
+				maintenanceHour: "Two",
+				ctx:             context.TODO(),
+			},
+			want:    integreatlyv1alpha1.PhaseFailed,
+			wantErr: true,
+		},
+		{
+			name: "failure reconciling strategy map",
+			fields: fields{
+				Config: nil,
+				ConfigManager: &config.ConfigReadWriterMock{
+					GetOperatorNamespaceFunc: func() string {
+						return "redhat-rhoam-operator"
+					},
+				},
+				installation: nil,
+				mpm:          nil,
+				log:          getLogger(),
+				Reconciler: resources.NewReconciler(&marketplace.MarketplaceInterfaceMock{}).
+					WithProductDeclaration(marketplace.ProductDeclaration{}),
+				recorder: nil,
+			},
+			args: args{
+				client:          moqclient.NewSigsClientMoqWithScheme(scheme, &configv1.Infrastructure{}),
+				maintenanceDay:  "4",
+				maintenanceHour: "2",
+				ctx:             context.TODO(),
+			},
+			want:    integreatlyv1alpha1.PhaseFailed,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &Reconciler{
+				Config:        tt.fields.Config,
+				ConfigManager: tt.fields.ConfigManager,
+				installation:  tt.fields.installation,
+				mpm:           tt.fields.mpm,
+				log:           tt.fields.log,
+				Reconciler:    tt.fields.Reconciler,
+				recorder:      tt.fields.recorder,
+			}
+			got, err := r.reconcileCloudResourceStrategies(tt.args.client, tt.args.maintenanceDay, tt.args.maintenanceHour)
```

Comment:
Unit tests likely need to be updated now that we've moved getting the addon parameters to this function 🤔

Reply:
Yes of course, apologies, will update.